sktime/forecasting/arch/_uarch.py

```diff
@@ -252,8 +252,8 @@ def _fit(self, y, X=None, fh=None):
             self._horizon = fh
 
         self._forecaster = _ARCH(
-            y=y,
-            x=X,
+            y=y.copy(deep=True),
+            x=X.copy(deep=True) if X is not None else X,
```

Comment:
why do we need to make copies?
Does `_ARCH` have side effects? If yes, I would report this as a bug in the arch package.

Reply:
![image](https://github.com/sktime/sktime/assets/70791259/904896b6-4a83-4987-a9f5-69684e1fa809)
```
import pandas as pd
from sktime.datasets import load_airline
y = load_airline()
y.name = None
print("Before fit: ", y.name, type(y.name))
from arch import arch_model
model = arch_model(y, rescale=True)
print("After fit: ", y.name, type(y.name))
```
Yes, bug need to be reported.

Reply:
is this only `ARCH`, or other estimators in general?

If only `ARCH`, we could overwrite the `name` at the end of `_fit`, instead of creating a full copy?

Reply:
Other estimators are fine, it's only `ARCH`. Let's overwrite the `name` at the end of `_fit` as you said.